src/popup/router/components/Modals/ConfirmTransactionSign.vue

```diff
@@ -130,45 +162,146 @@ export default {
       'pointers',
     ],
   }),
-  computed: mapGetters([
-    'getTxSymbol',
-    'getTxAmountTotal',
-    'getTxFee',
-    'getNameFee',
-    'isTxAex9',
-  ]),
+  computed: {
+    ...mapState(['sdk']),
+    ...mapState('fungibleTokens', ['availableTokens']),
+    ...mapGetters(['formatCurrency', 'account', 'activeNetwork']),
+    ...mapGetters([
+      'getTxSymbol',
+      'getTxAmountTotal',
+      'getTxFee',
+      'getNameFee',
+      'isTxAex9',
+      'getDexContracts',
+      'getTxDirection',
+    ]),
+    isAllowance() {
+      return FUNCTION_TYPE_DEX.allowance.includes(this.txFunction);
+    },
+    isSwap() {
+      return FUNCTION_TYPE_DEX.swap.includes(this.txFunction);
+    },
+    isPool() {
+      return FUNCTION_TYPE_DEX.pool.includes(this.txFunction);
+    },
+    isDex() {
+      return this.getDexContracts.router.includes(this.transaction?.contractId)
+        || this.getDexContracts.wae.includes(this.transaction?.contractId)
+        || this.isAllowance;
+    },
+    singleToken() {
+      return {
+        isReceived: this.getTxDirection(this.transaction) === 'received',
+        amount: this.getTxAmountTotal(this.transaction),
+        symbol: this.getTxSymbol(this.transaction),
+        isAex9: this.isTxAex9(this.transaction),
+        preventAmountConversion: true,
+        showFiat: true,
+      };
+    },
+    filteredTxFields() {
+      return this.TX_FIELDS.filter((field) => !!this.transaction[field]);
+    },
+    swapDirection() {
+      if (['swap_tokens_for_exact_tokens', 'swap_tokens_for_exact_ae', 'swap_ae_for_exact_tokens'].includes(this.txFunction)) return 'minReceived';
+      if (['swap_exact_tokens_for_tokens', 'swap_exact_ae_for_tokens', 'swap_exact_tokens_for_ae'].includes(this.txFunction)) return 'maxSpent';
+      return '';
+    },
+    completeTransaction() {
+      return { ...this.transaction, function: this.txFunction };
+    },
+  },
+  async mounted() {
+    if (this.transaction.contractId) {
+      try {
+        this.loading = true;
+        setTimeout(() => { this.loading = false; }, 20000);
+        await this.$watchUntilTruly(() => this.sdk);
```

Comment:
**fix** error on initing the popup (waiting for `sdk` being trully to get a `compilerApi`), was not loading any data at all
`TypeError: Cannot read properties of null (reading 'compilerApi')`